### PR TITLE
Add note about GCC 5 in re2 README, prereqs.rst

### PR DESCRIPTION
**Pull request by mppf**
* Updates prereqs.rst to note GCC 5 for C++11 dependency in third-party
* Clarifies some error wording in Regexp.chpl
* Updates third-party/re2/README to note C++11 dependency and suggest GCC 5 and say why.

Reviewed by @dmk42 and @ben-albrecht - thanks!

### CODE REVIEW

**Comment by mppf** [comment]
@dmk42 - how does this look?

**Comment by dmk42** [comment]
I agree with this change.  I would like to see an opinion from @ronawho .

**Comment by ronawho** [comment]
I think this needs to be somewhere way more visible like prereqs.rst

File: doc/rst/usingchapel/prereqs.rst [hunk 1]
```diff
@@ -29,6 +29,10 @@ about your environment for using Chapel:
     relatively recent versions of gcc/g++, clang, and compilers from
     Cray, Intel, and PGI.
 
+    * Note that you will need a C++11 compiler to build LLVM or regular
+      expression support (i.e.  CHPL_LLVM=llvm or CHPL_REGEXP=re2). If
+      GCC is used, we recommend GCC version 5 or newer for this purpose.
+
```
Comment (dmk42) [inline doc/rst/usingchapel/prereqs.rst]: Also, when gmp builds on chapcs, I see it compile in C++11 mode.  Should something about gmp be mentioned here?  I think on other systems it sometimes uses the system gmp and doesn't need C++11, so the situation might be complicated.

Comment (dmk42) [inline doc/rst/usingchapel/prereqs.rst]: Nope, it looks like that was my mistake.  I was misreading my log from the build.

**Comment by dmk42** [comment]
:+1:

File: modules/standard/Regexp.chpl [hunk 2]
```diff
@@ -42,9 +42,9 @@ in the Chapel distribution.
 
 .. note::
 
-  if re2 support is not enabled (which is the default), all features
-  described below will compile successfully, but will result in an internal
-  error at *run time*, saying "No Regexp Support".
+  if re2 support is not enabled (which is the case in quickstart
```
Comment (ben-albrecht) [inline modules/standard/Regexp.chpl]: Could possibly cross-reference `:ref:quickstart-with-other-shells` to make it clear what 'quickstart' is.

**Comment by mppf** [comment]
@ben-albrecht - it made more sense to me to link to the whole Quickstart document. Sound OK? Otherwise, I think we'd need to add a paragraph somewhere like "What is Quickstart mode?".

**Comment by ben-albrecht** [comment]
Sounds good! LGTM

**Pull request merged**